webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -1233,6 +1282,118 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
             zipOut.putNextEntry(null, parameters);
             zipOut.write(join(stripNewlines(scrambleRequest.getAllScrambles()), "\r\n").getBytes());
             zipOut.closeEntry();
+                
+            parameters.setFileNameInZip("Internationalization/");
+            
+            ArrayList<String> availableLanguages = new ArrayList<String>();
+            // TODO
+            // Jeremy, this is where the iteration over files can let internationalization easier to handle.
+            // The following lines must be replaced.
+            availableLanguages.add("MessagesBundle.properties");
+            availableLanguages.add("MessagesBundle_en_US.properties");
+            availableLanguages.add("MessagesBundle_pt_BR.properties");
+            
+            for (String item : availableLanguages) {
+                String[] temp = item.split("_|\\.");
+                
+                if (temp.length != 4){
+                    continue;
+                }
+                
+                String language = temp[1];
+                String country = temp[2];
+                
+                // fewest moves regular sheet
+                pdfFileName = "Internationalization/"+language+"_"+country+"_"+safeTitle+".pdf";
+                parameters.setFileNameInZip(pdfFileName);
+                zipOut.putNextEntry(null, parameters);
+
+                ByteArrayOutputStream pdfOut = new ByteArrayOutputStream();
+                Rectangle pageSize = PageSize.LETTER;
+                Document doc = new Document(pageSize, 0, 0, 75, 75);
+                PdfWriter docWriter = PdfWriter.getInstance(doc, pdfOut);
+
+                docWriter.setBoxSize("art", new Rectangle(36, 54, pageSize.getWidth()-36, pageSize.getHeight()-54));
+
+                doc.addCreationDate();
+                doc.addProducer();
+                if(globalTitle != null) {
+                    doc.addTitle(globalTitle);
+                }
+
+                doc.open();
+                for (int i=0; i<scrambleRequest.scrambles.length; i++){
+                    addFmcSolutionSheet(docWriter, doc, scrambleRequest, globalTitle, i, language, country);
```

Comment:
It looks painful to pass around language *and* country everywhere you go. How about just passing around a Locale object instead?

Reply:
Allright